sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -184,9 +188,16 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
 
     val cls = DataSource.lookupDataSource(source)
     if (classOf[DataSourceV2].isAssignableFrom(cls)) {
-      val options = new DataSourceV2Options(extraOptions.asJava)
+      val dataSource = cls.newInstance()
+      val options = dataSource match {
+        case cs: ConfigSupport =>
+          val confs = withSessionConfig(cs, sparkSession.sessionState.conf)
+          new DataSourceV2Options((confs ++ extraOptions).asJava)
```

Comment:
What happened if they have duplicate names?

Reply:
Good catch! Should the confs in the `extraOptions` have a higher priority? WDYT @cloud-fan ?

Reply:
yea `extraOptions` needs higher priority.